policy_integration_beta_test.go

```diff
@@ -0,0 +1,349 @@
+		Enforce: []*EnforcementOptions{
+			{
+				Path: String(".rego"),
+				Mode: EnforcementMode(EnforcementMandatory),
+			},
+		},
+	}
+	pTest3, pTestCleanup3 := createPolicyWithOptions(t, client, orgTest, opaOptions)
+	defer pTestCleanup3()
+
+	t.Run("without list options", func(t *testing.T) {
+		pl, err := client.Policies.List(ctx, orgTest.Name, nil)
+		require.NoError(t, err)
+		assert.Contains(t, pl.Items, pTest1)
+		assert.Contains(t, pl.Items, pTest2)
+		assert.Contains(t, pl.Items, pTest3)
+
+		assert.Equal(t, 1, pl.CurrentPage)
+		assert.Equal(t, 3, pl.TotalCount)
+	})
+
+	t.Run("with pagination", func(t *testing.T) {
+		// Request a page number which is out of range. The result should
+		// be successful, but return no results if the paging options are
+		// properly passed along.
+		pl, err := client.Policies.List(ctx, orgTest.Name, &PolicyListOptions{
+			ListOptions: ListOptions{
+				PageNumber: 999,
+				PageSize:   100,
+			},
+		})
+		require.NoError(t, err)
+
+		assert.Empty(t, pl.Items)
+		assert.Equal(t, 999, pl.CurrentPage)
+		assert.Equal(t, 3, pl.TotalCount)
+	})
+
+	t.Run("with search", func(t *testing.T) {
+		// Search by one of the policy's names; we should get only that policy
+		// and pagination data should reflect the search as well
+		pl, err := client.Policies.List(ctx, orgTest.Name, &PolicyListOptions{
+			Search: pTest1.Name,
+		})
+		require.NoError(t, err)
+
+		assert.Contains(t, pl.Items, pTest1)
+		assert.NotContains(t, pl.Items, pTest2)
+		assert.Equal(t, 1, pl.CurrentPage)
+		assert.Equal(t, 1, pl.TotalCount)
+	})
+
+	t.Run("with filter by kind", func(t *testing.T) {
+		pl, err := client.Policies.List(ctx, orgTest.Name, &PolicyListOptions{
+			Kind: OPA,
+		})
+		require.NoError(t, err)
+
+		assert.Contains(t, pl.Items, pTest3)
+		assert.NotContains(t, pl.Items, pTest1)
+		assert.NotContains(t, pl.Items, pTest2)
+		assert.Equal(t, 1, pl.CurrentPage)
+		assert.Equal(t, 1, pl.TotalCount)
+	})
+
+	t.Run("without a valid organization", func(t *testing.T) {
+		ps, err := client.Policies.List(ctx, badIdentifier, nil)
+		assert.Nil(t, ps)
+		assert.EqualError(t, err, ErrInvalidOrg.Error())
+	})
+}
+
+func TestPoliciesUpdate_Beta(t *testing.T) {
+	skipIfNotCINode(t)
+	skipIfFreeOnly(t)
+	skipIfBeta(t)
+
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	defer orgTestCleanup()
+
+	t.Run("with a new query", func(t *testing.T) {
+		options := PolicyCreateOptions{
+			Description: String("A sample policy"),
+			Kind:        OPA,
+			Query:       String("terraform.main"),
+			Enforce: []*EnforcementOptions{
+				{
+					Path: String(".rego"),
+					Mode: EnforcementMode(EnforcementMandatory),
+				},
+			},
+		}
+		pBefore, pBeforeCleanup := createUploadedPolicyWithOptions(t, client, true, orgTest, options)
+		defer pBeforeCleanup()
+
+		pAfter, err := client.Policies.Update(ctx, pBefore.ID, PolicyUpdateOptions{
```

Comment:
Since updating `query` is only supported for OPA kind and there is currently no validation for that in Policies.Update, would it be beneficial to add a test case that attempts to update the `query` when the kind is not OPA?

Reply:
good callout! :) Fixed!